src/applications/vaos/tests/new-appointment/components/CommunityCareLanguagePage.unit.spec.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { expect } from 'chai';
+import userEvent from '@testing-library/user-event';
+import { waitFor } from '@testing-library/dom';
+import { mockFetch, resetFetch } from 'platform/testing/unit/helpers';
+import {
+  createTestStore,
+  renderWithStoreAndRouter,
+  setTypeOfCare,
+  setTypeOfFacility,
+} from '../../mocks/setup';
+import CommunityCareLanguagePage from '../../../new-appointment/components/CommunityCareLanguagePage';
+
+const initialState = {
+  featureToggles: {
+    vaOnlineSchedulingCommunityCare: true,
+  },
+};
+
+let store;
+
+describe('VAOS integration: community care language preference page', () => {
+  beforeEach(async () => {
+    mockFetch();
+    store = createTestStore(initialState);
+    await setTypeOfCare(store, /primary care/i);
+    await setTypeOfFacility(store, /Community Care/i);
+  });
+  afterEach(() => resetFetch());
+
+  it('should show render language question and dropdown', async () => {
+    const screen = renderWithStoreAndRouter(<CommunityCareLanguagePage />, {
+      store,
+    });
+    expect(screen.baseElement).to.contain.text('Preferred language');
+    expect(screen.baseElement).to.contain.text(
```

Comment:
I think you would need to do `await waitFor` or a `findBy` query for this text, since it might not be shown on the first render and could result in a flaky test.

Same for most of the rest of the tests in here, too. We need to make sure to `await` for something within the form to render after the `renderWithStoreAndRouter` call.

Reply:
Added `waitFor`s after each render